.github/actions/npm-blacklist/checker.sh

```
@@ -56,44 +56,92 @@ fi
     if [[ "$line" =~ ^(@?[^@[:space:]]+)([@[:space:]]+(.+))?$ ]]; then
         pkg_name="${BASH_REMATCH[1]}"
         pkg_version="${BASH_REMATCH[3]:-}"
     else
-        echo "Warning: Invalid format in line: $line"
-        continue
+        echo "Warning: Invalid format in line: $line" >&2
+        return 0
     fi
     
     # Remove any trailing whitespace from version
     pkg_version=$(echo "$pkg_version" | sed 's/[[:space:]]*$//')
```

Shellcheck (0.11.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)